src/main/java/org/wmn4j/io/musicxml/MusicXmlWriterDom.java

```diff
@@ -480,8 +474,6 @@ private void handleMidMeasureClefChanges(
 				attributesElement.appendChild(clefElement);
 				measureElement.appendChild(attributesElement);
 
-				clefInEffect = undealtClefChanges.get(offset);
```

Comment:
I think this will lead to an additional clef attribute to be added to the beginning of the measure after a clef change has happened in the middle of the previous measure. Even though the output will be correct, it will contain unnecessary clef elements, so I will fix this. :)

Reply:
The last clef in effect in the previous measure is now used in checking for need to write the clef element into attributes.

Reply:
Great! :) That was all I got, so I'd say this is ready to be merged.